src/main/java/be/seeseemelk/mockbukkit/ServerMock.java

```diff
@@ -1399,15 +1404,17 @@ public World createWorld(@NotNull WorldCreator creator)
 	@Override
 	public boolean unloadWorld(String name, boolean save)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return unloadWorld(getWorld(name), save);
 	}
 
 	@Override
 	public boolean unloadWorld(World world, boolean save)
```

Comment:
Looking at Paper's implementation, there needs to be some logic added here.

- Overworld dimensions cannot be unloaded.
- Worlds with players cannot be unloaded.
- The `WorldUnloadEvent` should be called.

Reply:
The issue is uh, there isn't any default worlds loaded by mockbukkit in the first place, so there isnt a way to know which is the default overworld dimension.

Will implement the player and event call.

Also I realise.. there isn't `WorldLoadEvent` and `WorldInitEvent` being called for worldCreate method

Reply:
> there isn't any default worlds loaded by mockbukkit in the first place, so there isnt a way to know which is the default overworld dimension.

It isn't just the default overworld that it doesn't let you unload; It's any overworld world.
```java
if (handle.dimension() == net.minecraft.world.level.Level.OVERWORLD) {
    return false;
}
```

---

> Also I realise.. there isn't WorldLoadEvent and WorldInitEvent being called for worldCreate method

That'd be a bug, you can fix it in another PR if you'd like, otherwise we'll get around to it soon.

Reply:
hmmmm... Doing Bukkit#unloadWorld() on an actual server worlds on all worlds except the default world specific in server.properties `level-name`